lib/pages/task_list.dart

```diff
@@ -75,47 +73,64 @@ class _TaskListState extends State<TaskList> {
         context: context,
         builder: (context) {
           return Container(
+            key: const Key('edit_container'),
             padding: const EdgeInsets.symmetric(vertical: 20, horizontal: 60),
             color: bgColor,
             child: EditTask(
                 task: task,
                 onTaskUpdated: (updatedTask) {
-                  context
-                      .read<TaskCubit>()
-                      .updateTask(updatedTask: updatedTask);
+                  context.read<TaskCubit>().updateTask(task: updatedTask);
                 }),
           );
         });
   }
 
-  void _onTapCompleteIcon(Task task) {
+  void _onTapCompleteIcon(BuildContext context, Task task) {
     showDialog(
       context: context,
       builder: (context) => AlertDialog(
-          title: Text(
-            'Complete or Recycle Task?',
-            style: TextStyle(color: white),
-          ),
-          backgroundColor: bgColor,
-          surfaceTintColor: bgColor,
-          actions: [
-            TextButton(
-              child: Text(
-                'Complete',
-                style: TextStyle(color: white),
-              ),
-              onPressed: () => context.pop(),
+        key: const Key('complete_or_recycle'),
+        title: Text(
+          'Complete or Recycle Task?',
+          style: TextStyle(color: white),
+        ),
+        backgroundColor: bgColor,
+        surfaceTintColor: bgColor,
+        actions: [
+          TextButton(
+            key: const Key('incomplete_mark_complete'),
+            child: Text(
+              'Complete',
+              style: TextStyle(color: white),
             ),
-            TextButton(
-              child: Text(
-                'Recycle',
-                style: TextStyle(color: white),
-              ),
-              onPressed: () => context.pop(),
+            onPressed: () async {
+              context
+                  .read<TaskCubit>()
+                  .updateTask(task: task, newStatus: TaskStatus.completed)
+                  .then((result) {
+                context.read<TaskCubit>().readTasksByStatus(TaskStatus.todo);
+                context.pop();
+              });
+            },
```

Comment:
Consider refactoring this handler to a separate, private function

Reply:
done